test/org/zaproxy/zap/spider/parser/SpiderHtmlFormParserUnitTest.java

```diff
@@ -969,16 +981,131 @@ public void shouldSetValuesToFieldsWithNoValueWhenParsingPostForm() {
                         param("_week", formattedDate("yyyy-'W'ww", date)),
                         param("submit", "Submit")))));
     }
+    
+    @Test
+    public void shouldProvidedCorrectFormDataToValueGenerator() {
+        // Given
+        TestValueGenerator valueGenerator = new TestValueGenerator();
+        SpiderHtmlFormParser htmlParser = createSpiderHtmlFormParser(valueGenerator);
+        TestSpiderParserListener listener = createTestSpiderParserListener();
+        HttpMessage msg = createMessageWith("FormsForValueGenerator.html");
+        Source source = createSource(msg);
+        // When
+        boolean completelyParsed = htmlParser.parseResource(msg, source, BASE_DEPTH);
+        // Then
+        assertThat(valueGenerator.getFields(), hasSize(6));
+        assertThat(
+                valueGenerator.getFields().get(0),
+                is(equalTo(formField(
+                        "http://example.com/",
+                        // XXX The following should be: "http://example.org/post"
+                        null,
```

Comment:
Correct, that should use the value mentioned in the above comment. Sorry, for not being clear.